pkg/util/ranger/detacher.go

```diff
@@ -897,21 +911,29 @@ func (d *rangeDetacher) detachDNFCondAndBuildRangeForIndex(condition *expression
 				valInfo := extractValueInfo(item)
 				sameValue, err := isSameValue(d.sctx.TypeCtx, columnValues[0], valInfo)
 				if err != nil {
-					return nil, nil, nil, false, errors.Trace(err)
+					return nil, nil, nil, false, -1, errors.Trace(err)
 				}
 				if !sameValue {
 					columnValues[0] = nil
 				}
 			}
+			if minAccessConds == -1 || 1 < minAccessConds {
```

Comment:
```suggestion
			if minAccessConds == -1 || minAccessConds > 1 {
```
I would prefer this one. But I am OK with the current one.